crates/usvg/src/writer.rs

```diff
@@ -1228,3 +1395,173 @@ fn write_num(num: f32, buf: &mut Vec<u8>, precision: u8) {
 
     write!(buf, "{}", v).unwrap();
 }
+
+/// Write all of the tspan attributes except for baseline_shift and decorations.
+fn write_span(
+    is_clip_path: bool,
+    writer_context: &mut WriterContext,
+    xml: &mut XmlWriter,
+    text: &Text,
+    char_offset: &mut usize,
+    chunk: &TextChunk,
+    span: &TextSpan,
+) {
+    xml.start_svg_element(EId::Tspan);
+
+    xml.write_svg_attribute(AId::FontFamily, &span.font.families.join(", "));
+
+    match span.font.style {
+        FontStyle::Normal => {}
+        FontStyle::Italic => xml.write_svg_attribute(AId::FontStyle, "italic"),
+        FontStyle::Oblique => xml.write_svg_attribute(AId::FontStyle, "oblique"),
+    }
+
+    if span.font.weight != 400 {
+        xml.write_svg_attribute(AId::FontWeight, &span.font.weight);
+    }
+
+    xml.write_svg_attribute(AId::FontSize, &span.font_size);
+
+    match span.visibility {
+        Visibility::Visible => {}
+        Visibility::Hidden => xml.write_svg_attribute(AId::Visibility, "hidden"),
+        Visibility::Collapse => xml.write_svg_attribute(AId::Visibility, "collapse"),
+    }
+
+    if span.letter_spacing != 0.0 {
+        xml.write_svg_attribute(AId::LetterSpacing, &span.letter_spacing);
+    }
+
+    if span.word_spacing != 0.0 {
+        xml.write_svg_attribute(AId::WordSpacing, &span.word_spacing);
+    }
+
+    if let Some(text_length) = span.text_length {
+        xml.write_svg_attribute(AId::TextLength, &text_length);
+    }
+
+    if span.length_adjust == LengthAdjust::SpacingAndGlyphs {
+        xml.write_svg_attribute(AId::LengthAdjust, "spacingAndGlyphs");
+    }
+
+    if span.small_caps {
+        xml.write_svg_attribute(AId::FontVariant, "small-caps");
+    }
+
+    if span.paint_order == PaintOrder::StrokeAndFill {
+        xml.write_svg_attribute(AId::PaintOrder, "stroke fill");
+    }
+
+    if !span.apply_kerning {
+        xml.write_attribute_raw("style", |buf| {
+            buf.extend_from_slice("font-kerning:none".as_bytes())
+        });
+    }
+
+    match span.dominant_baseline {
```

Comment:
Better check for Auto first and then `match` can returns just a string. Kinda like:

```rust
let name = match span.dominant_baseline {
    DominantBaseline::UseScript => "use-script"
    // ...
};
xml.write_svg_attribute(AId::DominantBaseline, name);
```

Same with `AlignmentBaseline`. 

This would make the code much cleaner.

Reply:
So something like this? Is there a way to avoid the final unreachable?
```rust
if span.dominant_baseline != DominantBaseline::Auto {
        let name = match span.dominant_baseline {
            DominantBaseline::UseScript => "use-script",
            ...
            DominantBaseline::Auto => unreachable!()
        };
        xml.write_svg_attribute(AId::DominantBaseline, name);
    }
```

Or should I remove the if statement and just return "auto" for Auto?

Reply:
Just keep the `unreachable!()`. It's fine.